dvc/repo/__init__.py

```diff
@@ -239,7 +239,14 @@ def __str__(self):
     def index(self):
```

Comment:
Other suggestion that I have is, as we discussed `repo.index` is a full index, so an API to get a partial index probably should not be included inside `Index`, but should be at Repo level, something like `repo.index_view()`.

Since Index now is more like a dataclass, we could, in theory, create an Index with partially filled data.

```
def index_view(directory):
   return Index(stages_from_dir, metrics_from_dir, plots_from_dir, params_from_dir)
```
But the collection happens eagerly.

Reply:
Makes sense 👍